sql/core/src/test/java/test/org/apache/spark/sql/execution/sort/RecordBinaryComparatorSuite.java

```diff
@@ -321,4 +321,69 @@ public void testBinaryComparatorWhenOnlyTheLastColumnDiffers() throws Exception
 
     assert(compare(0, 1) < 0);
   }
+
+  @Test
+  public void testBinaryComparatorGiveSameResultWhenComparedByteByByteAndComparedByLong()
+      throws Exception {
+    int numFields = 1;
+
+    UnsafeRow row1 = new UnsafeRow(numFields);
```

Comment:
Do you need UnsafeRow in this test? the test case I posted seems fine by itself. It isn't specific to UnsafeRow, or at least, I'd make sure this test still triggers the problem after the change.

Reply:
test cast updated